synapseclient/api/agent_services.py

```diff
@@ -0,0 +1,182 @@
+"""This module is responsible for exposing the services defined at:
+<https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.AgentController>
+"""
+
+import json
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+async def register_agent(
+    cloud_agent_id: str,
+    cloud_alias_id: Optional[str] = None,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Registers an agent with Synapse OR gets existing agent registration.
+    Sends a request matching
+    <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistrationRequest.html>
+
+    Arguments:
+        cloud_agent_id: The cloud provider ID of the agent to register.
+        cloud_alias_id: The cloud provider alias ID of the agent to register.
+                        In the Synapse API, this defaults to 'TSTALIASID'.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The registered agent matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistration.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    request = {"awsAgentId": cloud_agent_id}
+    if cloud_alias_id:
+        request["awsAliasId"] = cloud_alias_id
+    return await client.rest_put_async(
+        uri="/agent/registration", body=json.dumps(request)
+    )
+
+
+async def get_agent(
+    registration_id: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Any]:
+    """
+    Gets information about an existing agent registration.
+
+    Arguments:
+        registration_id: The ID of the agent registration to get.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The requested agent registration matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistration.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(uri=f"/agent/registration/{registration_id}")
+
+
+async def start_session(
+    access_level: str,
+    agent_registration_id: str,
+    synapse_client: Optional["Synapse"] = None,
```

Comment:
I had been forcing `synapse_client` to be passed in by a kwargument. It would allow us to add new arguments to functions and force folks to re-order arguments:

```
async def start_session(
    access_level: str,
    agent_registration_id: str,
    *,
    synapse_client: Optional["Synapse"] = None,
```

Reply:
Updated in all agent_services functions